src/model/pane.ts

```diff
@@ -407,23 +419,28 @@ export class Pane implements IDestroyable {
 		return { minZOrder: minZOrder, maxZOrder: maxZOrder };
 	}
 
-	private _insertDataSource(source: IDataSource, preferredPosition: PreferredPriceScalePosition, zOrder: number): void {
+	private _insertDataSource(source: IDataSource, priceScaleId: string, zOrder: number): void {
 		let priceScale: PriceScale | null = null;
 
 		if (source instanceof PriceDataSource) {
-			priceScale = this._findSuitableScale(source, preferredPosition);
+			priceScale = this.priceScaleById(priceScaleId);
 		}
 
-		this._dataSources.push(source);
-		if (preferredPosition === 'overlay') {
-			this._overlaySources.push(source);
+		if (priceScale === null) {
+			priceScale = this._createPriceScale(priceScaleId, this._model.options().priceScale);
+
 		}
 
-		if (priceScale !== null) {
-			priceScale.addDataSource(source);
-			source.setPriceScale(priceScale);
+		this._dataSources.push(source);
+		if (priceScaleId !== 'left' && priceScaleId !== 'right') {
+			const overlaySources = this._overlaySources.get('priceScaleId') || [];
```

Comment:
```suggestion
			const overlaySources = this._overlaySources.get(priceScaleId) || [];
```

is it typo here? I think we need to add test for that.

Reply:
Yes. Added a test for this